src/admin/server.ts

```diff
@@ -1,41 +1,53 @@
-import { ApolloServer } from 'apollo-server-express';
+import { ApolloServer } from '@apollo/server';
+import { Server } from 'http';
 import { buildSubgraphSchema } from '@apollo/subgraph';
-import { Request } from 'express';
 import { typeDefsAdmin } from '../typeDefs';
 import { resolvers as adminResolvers } from './resolvers';
 import { errorHandler, sentryPlugin } from '@pocket-tools/apollo-utils';
+import { ApolloServerPluginLandingPageGraphQLPlayground } from '@apollo/server-plugin-landing-page-graphql-playground';
 import {
   ApolloServerPluginLandingPageDisabled,
-  ApolloServerPluginLandingPageGraphQLPlayground,
-} from 'apollo-server-core';
-import { ContextManager } from './context';
-
-// Function signature for context creator; primarily for
-// injecting test contexts
-interface ContextFactory {
-  (req: Request): ContextManager;
-}
+  ApolloServerPluginInlineTraceDisabled,
+  ApolloServerPluginUsageReportingDisabled,
+} from '@apollo/server/plugin/disabled';
+import { ApolloServerPluginInlineTrace } from '@apollo/server/plugin/inlineTrace';
+import { ApolloServerPluginDrainHttpServer } from '@apollo/server/plugin/drainHttpServer';
+import { IAdminContext } from './context';
 
 /**
  * Sets up and configures an ApolloServer for the application.
  * @param contextFactory function for creating the context with
  * every request
  * @returns ApolloServer
  */
-export function getServer(contextFactory: ContextFactory): ApolloServer {
-  return new ApolloServer({
+export function getAdminServer(
+  httpServer: Server
+): ApolloServer<IAdminContext> {
+  const defaultPlugins = [
+    sentryPlugin,
+    ApolloServerPluginDrainHttpServer({ httpServer }),
+  ];
+  const prodPlugins = [
+    ApolloServerPluginLandingPageDisabled(),
+    ApolloServerPluginInlineTrace(),
+  ];
+  const nonProdPlugins = [
```

Comment:
nit: thoughts on using `devPlugins` ? Although, `nonProdPlugins` future proofs us for when / if we get a staging environment 😄

Reply:
I honestly hope none of this boilerplate is still in individual repos by the time we get around to doing that.  So much gets left behind as new changes are introduced to apollo-utils and one off QoL changes to individual repos.

With that said, would it really have a 3rd independent set of plugins? I feel like it either matches one of these, and should probably be changed to a function that pattern matches by environment rather than adding more and more ternary branches if not.